manual/embedded-examples/src/docs/dev/procedures.asciidoc

```diff
@@ -0,0 +1,495 @@
+     <scope>provided</scope>
+   </dependency>
+----
+
+Next, the dependencies necessary for testing the procedure are added:
+
+* Neo4j harness, a utility that allows for starting a light-weight Neo4j instance.
+  It is used to start Neo4j with a specific procedure deployed, which greatly simplifies testing.
+* The Neo4j Java driver, used to send cypher statements that call the procedure.
+* JUnit, a common Java test framework.
+
+[source, xml]
+----
+   <dependency>
+     <groupId>org.neo4j.test</groupId>
+     <artifactId>neo4j-harness</artifactId>
+     <version>${neo4j.version}</version>
+     <scope>test</scope>
+   </dependency>
+
+   <dependency>
+     <groupId>org.neo4j.driver</groupId>
+     <artifactId>neo4j-java-driver</artifactId>
+     <version>1.0-SNAPSHOT</version>
+     <scope>test</scope>
+   </dependency>
+
+   <dependency>
+     <groupId>junit</groupId>
+     <artifactId>junit</artifactId>
+     <version>4.12</version>
+     <scope>test</scope>
+   </dependency>
+----
+
+After dependencies are defined the steps that Maven will go through to build the project.
+The goal is first to _compile_ the source, then to _package_ it in a JAR that can be deployed to a Neo4j instance.
+
+[NOTE]
+--
+Procedures require at least Java 8, so the version `1.8` should be defined as the _source_ and _target version_ in the configuration for the Maven compiler plugin.
+--
+
+The Maven shade plugin is used to _package_ the compiled procedure.
+It also includes all dependencies of the procedure, unless the dependency scope is set to _test_ or _provided_.
+
+Once the procedure is provided in the `plugins` directory of each Neo4j instance and the instances have restarted, the procedure is available for use.
+
+[source, xml]
+----
+  <build>
+   <plugins>
+     <plugin>
+       <artifactId>maven-compiler-plugin</artifactId>
+       <configuration>
+         <source>1.8</source>
+         <target>1.8</target>
+       </configuration>
+     </plugin>
+     <plugin>
+       <artifactId>maven-shade-plugin</artifactId>
+       <executions>
+         <execution>
+           <phase>package</phase>
+           <goals>
+             <goal>shade</goal>
+           </goals>
+         </execution>
+       </executions>
+     </plugin>
+   </plugins>
+  </build>
+----
+
+Until the GA release of Neo4j 3.0, the dependency on Neo4j requires that a _snapshot repository_ is configured.
+This repository is where Maven will find the latest build of Neo4j to use as a dependency.
+
+[source, xml]
+----
+  <repositories>
+   <repository>
+     <id>neo4j-snapshot-repository</id>
+     <name>Maven 2 snapshot repository for Neo4j</name>
+     <url>http://m2.neo4j.org/content/repositories/snapshots</url>
+     <snapshots><enabled>true</enabled></snapshots>
+     <releases><enabled>false</enabled></releases>
+   </repository>
+ </repositories>
+</project>
+----
+
+
+=== Writing integration tests
+
+In the test dependencies are included Neo4j Harness and JUnit.
```

Comment:
Do you mean _The test dependencies include Neo4j Harness and JUnit._?
The phrasing makes it a bit tricky to parse.